lib/sinatra/base.rb

```diff
@@ -188,7 +187,7 @@ def finish
         headers["Content-Length"] = body.map(&:bytesize).reduce(0, :+).to_s
       end
 
-      [status.to_i, headers, result]
+      [status, headers, result]
```

Comment:
I can't say that this change is good.
For example, we can assign a value directly toresponse.status`.

In this case, it will be returned as a response as it is. Of course, in other parts of sinatra, some codes are written on the assumption that the status is integer. ref: https://github.com/sinatra/sinatra/blob/6d06e4907ee39bfafa18318b6567d43170f85ef8/lib/sinatra/base.rb#L607

And these conversions are confusing to the user, so the original code is not good.
Therefore, your point of view is not wrong.

However, I don't see the tremendous benefit of changing this. I think it makes sense to include such changes in major version upgrades.

Reply:
I think remove checking type of status in `finish` method brings clarity to the overall codebase. 

When I read the `Sinatra` source code, I found the all the `status`es are used base on the assumption that `status` is an int (include `Sinatra::Response`'s superclass `Rack::Response`)  except `finish` method.  Then I checked `finish` method and didn't find any particularity from this method. So I changed  two `status.to_i` and `status.to_i / 100 == 1` here.

Assign a value(not a int) directly to `response.status` will cause unexpected result while calling other methods in most cases. In my option, my change to `status` in `finish` sticks to the Sinatra assumption of type of `status`.

Reply:
>Assign a value(not a int) directly to response.status will cause unexpected result while calling other methods in most cases. 

OK, the corner cases can be ignored as per the our current code base